packages/itwin/ec3-widget/src/components/EC3/EC3AuthCallback.ts

```diff
@@ -16,27 +15,34 @@ import type { EC3Token } from "./EC3Token";
 *   });
 * } else {
 */
-export function handleEC3AuthCallback(props: EC3ConfigProps) {
-  const MILI_SECONDS = 1000;
-  const EXPIRATRION_REDUCTION = 15 * 60;
+
+export interface EC3AuthCallbackConfigProps {
+  clientId: EC3ConfigPropsWithRedirectUri["clientId"];
+  redirectUri: EC3ConfigPropsWithRedirectUri["redirectUri"];
+  ec3Uri?: EC3ConfigPropsWithRedirectUri["ec3Uri"];
+}
+
+export function handleEC3AuthCallback(ec3Config: EC3AuthCallbackConfigProps, source: string = "ec3-auth") {
+  const MILLI_SECONDS = 1000;
+  const EXPIRATION_REDUCTION = 15 * 60;
+  const ec3Uri = ec3Config.ec3Uri ?? "https://buildingtransparency.org/";
```

Comment:
Duplication of code from the `EC3Config` class.

Reply:
The fallback logic is still duplicated.

Reply:
Sorry, I hadn't made the change. Here it is.